pkg/clients/clients.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return nil, err
+	}
+	rm, err := apiutil.NewDynamicRESTMapper(cc.KubeConfig, httpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	opClient, err := NewOperatorClient(cc.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	configClient, err := configclient.NewForConfig(cc.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Clients{
+		KubeClient:            kubeClient,
+		APIExtensionsClient:   apiExtensionsClient,
+		DynamicClient:         dynClient,
+		RESTMapper:            rm,
+		OperatorClient:        opClient,
+		ConfigClient:          configClient,
+		KubeInformerFactory:   informers.NewSharedInformerFactory(kubeClient, defaultResyncPeriod),
+		ConfigInformerFactory: configinformer.NewSharedInformerFactory(configClient, defaultResyncPeriod),
+	}, nil
+}
+
+func (c *Clients) ClientHolder() *resourceapply.ClientHolder {
+	return resourceapply.NewClientHolder().
+		WithKubernetes(c.KubeClient).
+		WithDynamicClient(c.DynamicClient).
+		WithAPIExtensionsClient(c.APIExtensionsClient)
+}
+
+func NewOperatorClient(config *rest.Config) (*OperatorClient, error) {
+	client, err := operatorclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	factory := operatorinformers.NewSharedInformerFactory(client, defaultResyncPeriod)
+
+	return &OperatorClient{
+		client:    client,
+		informers: factory,
+	}, nil
+}
+
+var _ v1helpers.OperatorClientWithFinalizers = &OperatorClient{}
+
+const (
+	olmSingletonName = "cluster"
+	fieldManager     = "cluster-olm-operator"
+)
+
+type OperatorClient struct {
+	client    operatorclient.Interface
+	informers operatorinformers.SharedInformerFactory
+}
+
+func (o OperatorClient) Informer() cache.SharedIndexInformer {
+	return o.informers.Operator().V1alpha1().OLMs().Informer()
+}
+
+func (o OperatorClient) GetObjectMeta() (*metav1.ObjectMeta, error) {
+	olm, err := o.client.OperatorV1alpha1().OLMs().Get(context.TODO(), olmSingletonName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &olm.ObjectMeta, nil
+}
+
+func (o OperatorClient) GetOperatorState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, string, error) {
+	olm, err := o.informers.Operator().V1alpha1().OLMs().Lister().Get(olmSingletonName)
+	if err != nil {
+		return nil, nil, "", err
+	}
+
+	return &olm.Spec.OperatorSpec, &olm.Status.OperatorStatus, olm.ResourceVersion, nil
+}
+
+func (o OperatorClient) UpdateOperatorSpec(ctx context.Context, oldResourceVersion string, in *operatorv1.OperatorSpec) (*operatorv1.OperatorSpec, string, error) {
+	olm := &operatorv1alpha1.OLM{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            olmSingletonName,
+			ResourceVersion: oldResourceVersion,
+		},
+		Spec: operatorv1alpha1.OLMSpec{OperatorSpec: *in},
+	}
+	out, err := o.client.OperatorV1alpha1().OLMs().Update(ctx, olm, metav1.UpdateOptions{FieldManager: fieldManager})
+	if err != nil {
+		return nil, "", err
+	}
+	return &out.Spec.OperatorSpec, out.ResourceVersion, nil
```

Comment:
Probably safer to do like CSO:
1. Get original from informer cache
2. Deep copy
3. Set copy.ResourceVersion from param
4. Set copy.spec.OperatorSpec from param
5. Issue Update

Otherwise you can lose annotations/labels